internal/session/testing.go

```diff
@@ -135,7 +135,7 @@ func TestSessionParams(t testing.TB, conn *db.DB, wrapper wrapping.Wrapper, iamR
 	sets := static.TestSets(t, conn, cats[0].PublicId, 1)
 	_ = static.TestSetMembers(t, conn, sets[0].PublicId, hosts)
 
-	tcpTarget := tcp.TestTarget(ctx, t, conn, proj.PublicId, "test target")
+	tcpTarget := tcp.TestTarget(ctx, t, conn, proj.PublicId, "test target", target.WithSessionConnectionLimit(1))
```

Comment:
Without adding this option we get errors where sessions aren't progressing from `pending`. I will do some more digging, but "for now" this fixes the test errors.

Reply:
Probably good to get this figured out before release as we've had weird issues with session state management before...

Reply:
Yeah, I will keep looking

Reply:
Turns out that Boundary was doing the right thing here and not terminating sessions without a connection limit. The tests that expect a single connection limit have been updated to have this set explicitly on the target.